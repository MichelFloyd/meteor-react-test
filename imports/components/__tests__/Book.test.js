import React from 'react';
import renderer from 'react-test-renderer';
import Book from '../Book';
import books from './fake-book-data';

const clickFn = jest.fn();

test('provides book ID when clicked', () => {
  const component = renderer.create(
    <Book book={books[0]} onClick={clickFn} />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.props.onClick();
  expect(clickFn.mock.calls.length).toBe(1);
  expect(clickFn.mock.calls[0][0]).toBe('abc123');
});
