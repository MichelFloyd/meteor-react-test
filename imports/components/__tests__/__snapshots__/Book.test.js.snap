// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`provides book ID when clicked 1`] = `
<div
  className="root-0-0"
  onClick={[Function]}
>
  <div>
    <div
      className="root-0-2"
      style={
        Object {
          "height": 187,
          "width": 128,
        }
      }
    >
      <img
        className="image-0-5"
        height={187}
        src="http://placehold.it/128x187"
        title="Not a real book"
        width={128}
      />
      <div
        className="reflection-0-6"
      >
        <img
          className="flippedImage-0-3"
          height={187}
          src="http://placehold.it/128x187"
          width={128}
        />
        <div
          className="reflectionOverlay-0-8"
          style={
            Object {
              "height": 187,
              "width": 128,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
